keyboards/lfkeyboards/lfk87/keymaps/via/keymap.c

```diff
@@ -0,0 +1,120 @@
+    XXXXXXX, XXXXXXX, KC_PGUP, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, KC_UP,   XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,           _______, _______, _______, \
+    _______, KC_HOME, KC_PGDN, KC_END,  XXXXXXX, XXXXXXX, XXXXXXX, KC_LEFT, KC_DOWN, KC_RGHT, XXXXXXX, XXXXXXX,          XXXXXXX,                                      \
+    _______, XXXXXXX, KC_DEL,  XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, KC_MUTE, KC_VOLD, KC_VOLU, XXXXXXX,                   TG(SETTINGS),               _______,          \
+    _______, _______, _______,                            KC_PGDN,                            _______, _______, _______, _______,           _______, _______, _______  \
+  ),
+
+  /* Keymap SETTINGS: Settings Layer
+   * ,------------------------------------------------------------.  ,-------------.
+   * |Esc  |f1| f2| f3| f4|  | f5| f6| f7| f8|   | f9|f10|f11|f12 |  |Prnt|ScLk|Paus|
+   * |------------------------------------------------------------|  |--------------|
+   * |FN0 |   |   |   |   |   |   |   |   |  |   |BL-|BL+|BL Togl |  |RGB Tog |Val+|
+   * |------------------------------------------------------------|  |-------------|
+   * |MuMode|   |   |   |   |   |   |   |   |   |   |   |  |LEDTst|  |RGB Mode|Val-|
+   * |------------------------------------------------------------|  `-------------'
+   * |AudTgl |Hz+|MS+|   |   |   |   |   |   |   |   |   |  RST   |
+   * |------------------------------------------------------------|       ,----.
+   * |ClickTgl|Hz-|MS-|   |   |   |   |   |   |   |   |Layer Clr  |       |Hue+|
+   * |------------------------------------------------------------------------------.
+   * |    |    |    |                        |    |    |    |     |  |Sat-|Hue-|Sat+|
+   * `------------------------------------------------------------------------------'
+   */
+  [SETTINGS] = LAYOUT_tkl_ansi(
+    KC_FN0,           XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, BL_DEC,       XXXXXXX, XXXXXXX, XXXXXXX, \
+    KC_FN0,  XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, BL_DEC,  BL_INC,  BL_TOGG,      RGB_TOG, RGB_VAI, XXXXXXX, \
+    MU_MOD,  XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, KC_FN2,       RGB_MOD, RGB_VAD, XXXXXXX, \
+    AU_TOG,  KC_FN1,  KC_FN3,  XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,          RESET,                                   \
+    KC_FN5,  KC_FN2,  KC_FN4,  XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, MU_TOG,  XXXXXXX, XXXXXXX, XXXXXXX,                   XXXXXXX,               RGB_HUI,          \
+    XXXXXXX, XXXXXXX, XXXXXXX,                            KC_FN12,                            XXXXXXX, XXXXXXX, XXXXXXX, _______,      RGB_SAD, RGB_HUD, RGB_SAI  \
+  ),
+};
+
+const uint16_t PROGMEM fn_actions[] = {
```

Comment:
Not sure how I feel about this. It's part of how the LF boards work currently, but `fn_actions` is deprecated, so it may be that we need to refactor this stuff before any new keymaps can be added. I also don't know if VIA supports the FN keys.

Reply:
Hey man, thanks for your comments here!
I don't know if this can be removed or not, but it seems to me that VIA can recognize this fn actions by default.
I also don't know if this is necessary for this PCB as I really never used it so I don't mind removing it completely if it means we can merge it quicker.

Reply:
VIA does not support fn_actions keycodes.